
# Data Analysis for cost of care in healthcare
_by Hafsa Laeeque_

---
The task is to<br>
  > 1) analyze the **clinical and financial data** of patients hospitalized for a _certain condition_.<br>
    ~~2) join data given in the different tables.~~<br>
    3) find insights about **drivers of cost of care**.<br>
    4) document _approach, results and insights_ using [slides](https://docs.google.com/presentation/d/1-gYni51iGkYh4OCCr-BYKAV3YqUH4OzQOUAtz5MkrEg/edit?usp=sharing) and a [document](https://docs.google.com/document/d/1fQB0AP2ue_zKVUUAHx_1sJ626okOMtm_IdLOnw-9__A/edit?usp=sharing), both of which should have a similar narrative.<br>

---
We have cleaned and combined our data in [part 1](https://github.com/hafsalaeeque/cost-in-healthcare-DS-proj/blob/master/Analysis%20for%20healthcare%20-%20Part%201.ipynb). Now we will load our refined dataset and draw insights.

## Analysing the datasets
### Import packages

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
from datetime import datetime, timedelta 
%matplotlib inline

import warnings
warnings.filterwarnings("ignore")

In [3]:
#display columns and rows
pd.options.display.max_columns = 50
pd.options.display.max_rows = 4000

### (A) Load the datasets
A1 - Let's load the first dataset `bill_amount.csv`.

In [17]:
df = pd.read_csv('clean_datasets/patients.csv')

In [18]:
df.shape

(3400, 35)

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3400 entries, 0 to 3399
Data columns (total 35 columns):
Unnamed: 0            3400 non-null int64
patient_id            3400 non-null object
gender                3400 non-null object
race                  3400 non-null object
resident_status       3400 non-null object
date_of_birth         3400 non-null object
date_of_admission     3400 non-null object
date_of_discharge     3400 non-null object
medical_history_1     3400 non-null int64
medical_history_2     3167 non-null float64
medical_history_3     3400 non-null float64
medical_history_4     3400 non-null int64
medical_history_5     3096 non-null float64
medical_history_6     3400 non-null int64
medical_history_7     3400 non-null int64
preop_medication_1    3400 non-null int64
preop_medication_2    3400 non-null int64
preop_medication_3    3400 non-null int64
preop_medication_4    3400 non-null int64
preop_medication_5    3400 non-null int64
preop_medication_6    3400 non-null int6

In [7]:
df.head()

Unnamed: 0                        patient_id  gender     race  \
0           0  fa2d818b2261e44e30628ad1ac9cc72c  Female   Indian   
1           1  5b6477c5de78d0b138e3b0c18e21d0ae  Female  Chinese   
2           2  320aa16c61937447fd6631bf635e7fde    Male  Chinese   
3           3  320aa16c61937447fd6631bf635e7fde    Male  Chinese   
4           4  c7f3881684045e6c49020481020fae36    Male    Malay   

  resident_status date_of_birth date_of_admission date_of_discharge  \
0     Singaporean    1971-05-14        2014-09-20        2014-10-04   
1     Singaporean    1976-02-18        2014-08-08        2014-08-18   
2     Singaporean    1982-07-03        2011-03-10        2011-03-20   
3     Singaporean    1982-07-03        2013-07-02        2013-07-14   
4     Singaporean    1947-06-15        2015-10-28        2015-11-10   

   medical_history_1  medical_history_2  medical_history_3  medical_history_4  \
0                  0                0.0                0.0                  0   
1                  0                0.0                0.0                  0   
2                  0                0.0                0.0                  0   
3                  0                0.0                0.0                  1   
4                  0                1.0                0.0                  0   

   medical_history_5  medical_history_6  medical_history_7  \
0                0.0                  0                  1   
1                NaN                  0                  1   
2                0.0                  1                  0   
3                NaN                  0                  0   
4                0.0                  0                  0   

   preop_medication_1  preop_medication_2  preop_medication_3  \
0                   1                   0                   1   
1                   0                   0                   0   
2                   0                   1                   1   
3                   1                   1                   0   
4                   0                   1                   0   

   preop_medication_4  preop_medication_5  preop_medication_6  symptom_1  \
0                   0                   1                   1          1   
1                   1                   1                   1          1   
2                   1                   1                   0          0   
3                   1                   0                   1          1   
4                   0                   1                   1          1   

   symptom_2  symptom_3  symptom_4  symptom_5  lab_result_1  lab_result_2  \
0          1          0          0          1          14.2          27.0   
1          1          1          1          1          14.3          27.2   
2          0          1          1          1          14.4          26.6   
3          0          0          1          0          18.0          26.6   
4          1          1          1          1          14.9          26.7   

   lab_result_3  weight  height  freq               days_admitted   age  \
0          81.0    68.1   161.0     1  14 days 00:00:00.000000000  47.0   
1          92.0    76.5   151.0     1  10 days 00:00:00.000000000  42.0   
2          92.0    69.7   175.0     2  10 days 00:00:00.000000000  36.0   
3          92.0    66.7   175.0     2  12 days 00:00:00.000000000  36.0   
4          97.0    96.9   180.0     1  13 days 00:00:00.000000000  71.0   

     amount  
0  19564.17  
1  20465.40  
2  14435.91  
3   6578.39  
4  45452.31

In [12]:
df.drop(['Unnamed: 0'], axis=1, inplace =True)

In [13]:
df.head()

patient_id  gender     race resident_status  \
0  fa2d818b2261e44e30628ad1ac9cc72c  Female   Indian     Singaporean   
1  5b6477c5de78d0b138e3b0c18e21d0ae  Female  Chinese     Singaporean   
2  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
3  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
4  c7f3881684045e6c49020481020fae36    Male    Malay     Singaporean   

  date_of_birth date_of_admission date_of_discharge  medical_history_1  \
0    1971-05-14        2014-09-20        2014-10-04                  0   
1    1976-02-18        2014-08-08        2014-08-18                  0   
2    1982-07-03        2011-03-10        2011-03-20                  0   
3    1982-07-03        2013-07-02        2013-07-14                  0   
4    1947-06-15        2015-10-28        2015-11-10                  0   

   medical_history_2  medical_history_3  medical_history_4  medical_history_5  \
0                0.0                0.0                  0                0.0   
1                0.0                0.0                  0                NaN   
2                0.0                0.0                  0                0.0   
3                0.0                0.0                  1                NaN   
4                1.0                0.0                  0                0.0   

   medical_history_6  medical_history_7  preop_medication_1  \
0                  0                  1                   1   
1                  0                  1                   0   
2                  1                  0                   0   
3                  0                  0                   1   
4                  0                  0                   0   

   preop_medication_2  preop_medication_3  preop_medication_4  \
0                   0                   1                   0   
1                   0                   0                   1   
2                   1                   1                   1   
3                   1                   0                   1   
4                   1                   0                   0   

   preop_medication_5  preop_medication_6  symptom_1  symptom_2  symptom_3  \
0                   1                   1          1          1          0   
1                   1                   1          1          1          1   
2                   1                   0          0          0          1   
3                   0                   1          1          0          0   
4                   1                   1          1          1          1   

   symptom_4  symptom_5  lab_result_1  lab_result_2  lab_result_3  weight  \
0          0          1          14.2          27.0          81.0    68.1   
1          1          1          14.3          27.2          92.0    76.5   
2          1          1          14.4          26.6          92.0    69.7   
3          1          0          18.0          26.6          92.0    66.7   
4          1          1          14.9          26.7          97.0    96.9   

   height  freq               days_admitted   age    amount  
0   161.0     1  14 days 00:00:00.000000000  47.0  19564.17  
1   151.0     1  10 days 00:00:00.000000000  42.0  20465.40  
2   175.0     2  10 days 00:00:00.000000000  36.0  14435.91  
3   175.0     2  12 days 00:00:00.000000000  36.0   6578.39  
4   180.0     1  13 days 00:00:00.000000000  71.0  45452.31

The first excel file, `bill_amt.csv`, does not have any null values. The data shows the cost incurred for each unique bill.

---
A2 - Let's load the second dataset `bill_id.csv`.

In [8]:
bill_id_df = pd.read_csv('datasets/bill_id.csv')

In [9]:
bill_id_df.shape

(13600, 3)

In [10]:
bill_id_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13600 entries, 0 to 13599
Data columns (total 3 columns):
bill_id              13600 non-null int64
patient_id           13600 non-null object
date_of_admission    13600 non-null object
dtypes: int64(1), object(2)
memory usage: 318.8+ KB


In [11]:
bill_id_df.tail()

bill_id                        patient_id date_of_admission
13595  1641053864  a4c61deaa9ce86b4d2289eab6128b872        2015-12-28
13596  6956955826  ac52a32f8ce8c46d82df2d72052ae5a9        2015-12-28
13597  1399259594  4f67a54ab205cc9e7e2b0a4ee08e4fba        2015-12-28
13598  9243628699  a4c61deaa9ce86b4d2289eab6128b872        2015-12-28
13599  4808173213  a4c61deaa9ce86b4d2289eab6128b872        2015-12-28

In [12]:
bill_id_df.bill_id.nunique()

13600

The second excel file, `bill_id.csv`, does not have any null values. The data shows the _date_ for when a patient was admitted in the hospital for each unique bill. This could be joined with our first dataset so we can observe the cost incurred for each patient.

---
A3 - Let's load the third dataset `clinical_data.csv`.

In [13]:
data = pd.read_csv('datasets/clinical_data.csv')

In [14]:
data.shape

(3400, 26)

In [15]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3400 entries, 0 to 3399
Data columns (total 26 columns):
id                    3400 non-null object
date_of_admission     3400 non-null object
date_of_discharge     3400 non-null object
medical_history_1     3400 non-null int64
medical_history_2     3167 non-null float64
medical_history_3     3400 non-null object
medical_history_4     3400 non-null int64
medical_history_5     3096 non-null float64
medical_history_6     3400 non-null int64
medical_history_7     3400 non-null int64
preop_medication_1    3400 non-null int64
preop_medication_2    3400 non-null int64
preop_medication_3    3400 non-null int64
preop_medication_4    3400 non-null int64
preop_medication_5    3400 non-null int64
preop_medication_6    3400 non-null int64
symptom_1             3400 non-null int64
symptom_2             3400 non-null int64
symptom_3             3400 non-null int64
symptom_4             3400 non-null int64
symptom_5             3400 non-null int64
lab

In [16]:
data.isnull().sum().sum()

537

In [17]:
data.head(6)

id date_of_admission date_of_discharge  \
0  1d21f2be18683991eb93d182d6b2d220        2011-01-01        2011-01-11   
1  62bdca0b95d97e99e1c712048fb9fd09        2011-01-01        2011-01-11   
2  c85cf97bc6307ded0dd4fef8bad2fa09        2011-01-02        2011-01-13   
3  e0397dd72caf4552c5babebd3d61736c        2011-01-02        2011-01-14   
4  94ade3cd5f66f4584902554dff170a29        2011-01-08        2011-01-16   
5  59e07adc2dbc5f70131f57d003610d74        2011-01-07        2011-01-17   

   medical_history_1  medical_history_2 medical_history_3  medical_history_4  \
0                  0                1.0                 0                  0   
1                  0                0.0                 0                  0   
2                  0                0.0                 0                  0   
3                  0                1.0                No                  0   
4                  0                0.0                No                  0   
5                  0                NaN                No                  0   

   medical_history_5  medical_history_6  medical_history_7  \
0                0.0                  0                  0   
1                0.0                  0                  0   
2                0.0                  0                  0   
3                0.0                  1                  1   
4                0.0                  1                  1   
5                NaN                  0                  0   

   preop_medication_1  preop_medication_2  preop_medication_3  \
0                   1                   0                   1   
1                   0                   1                   1   
2                   0                   1                   1   
3                   1                   0                   1   
4                   0                   0                   0   
5                   0                   1                   1   

   preop_medication_4  preop_medication_5  preop_medication_6  symptom_1  \
0                   0                   0                   1          0   
1                   1                   1                   0          0   
2                   1                   1                   1          1   
3                   0                   0                   1          1   
4                   0                   1                   0          0   
5                   1                   1                   1          1   

   symptom_2  symptom_3  symptom_4  symptom_5  lab_result_1  lab_result_2  \
0          0          0          1          1          13.2          30.9   
1          0          1          1          1          13.8          22.6   
2          1          1          1          0          11.2          26.2   
3          1          1          1          1          13.3          28.4   
4          1          0          1          0          12.0          27.8   
5          0          1          1          1          15.8          31.0   

   lab_result_3  weight  height  
0         123.0    71.3   161.0  
1          89.0    78.4   160.0  
2         100.0    72.0   151.0  
3          76.0    64.4   152.0  
4          87.0    55.6   160.0  
5          75.0    78.8   169.0

The third excel file, `clinical_data.csv` has 25 types of clinical data entries of 3400 patients.<br>
- It covers the
    - the date as to when the patient was admitted and discharged,
    - height of patient,
    - weight of patient,
    - if patient has any/all 7 medical histories, 
    - if patient has had to take any/all the 6 preoperation medication, 
    - if patient showed any/all the 5 symptoms and
    - 3 lab results of patient<br>
    
- The `id` column of the dataset seems to be the `patient_id` column in the `bill_id_df` datset.<br>
- It has **537 null values** in the 2nd and 5th medical history data.<br>
- The dataset needs to be cleaned as there are
     - string objects like "No" in the 3rd medical history column and<br>
     - null values.<br>
     
Considering that we had 13,600 unique patients in the `bill_id_df` dataset, the `clinical_data.csv` has only 25% of these patients' clinical data. 

---
A4 - Let's load the last dataset `demographics.csv`.

In [18]:
demographic = pd.read_csv('datasets/demographics.csv')

In [19]:
demographic.shape

(3000, 5)

In [20]:
demographic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 5 columns):
patient_id         3000 non-null object
gender             3000 non-null object
race               3000 non-null object
resident_status    3000 non-null object
date_of_birth      3000 non-null object
dtypes: object(5)
memory usage: 117.3+ KB


In [21]:
demographic.head(10)

patient_id  gender     race    resident_status  \
0  fa2d818b2261e44e30628ad1ac9cc72c  Female   Indian        Singaporean   
1  5b6477c5de78d0b138e3b0c18e21d0ae       f  Chinese  Singapore citizen   
2  320aa16c61937447fd6631bf635e7fde    Male  Chinese  Singapore citizen   
3  c7f3881684045e6c49020481020fae36    Male    Malay  Singapore citizen   
4  541ad077cb4a0e64cc422673afe28aef       m  Chinese        Singaporean   
5  cf280265a73331d6cad35b4800e96abf  Female  Chinese                 PR   
6  94f7d3a8a4d6bb14859b64c3f03e4a6c       m    Malay        Singaporean   
7  43dfbeb8d76f3b00b8fa7a49e5a3eb6f       f  chinese        Singaporean   
8  2882e70ff56c2600bbbbb855fcfa96b9    Male  Chinese        Singaporean   
9  36e65f14c328fef0b02aa7d4047c6f74  Female  Chinese  Singapore citizen   

  date_of_birth  
0    1971-05-14  
1    1976-02-18  
2    1982-07-03  
3    1947-06-15  
4    1970-12-12  
5    1966-12-01  
6    1975-09-14  
7    1974-03-04  
8    1969-04-22  
9    1976-10-24

The last excel file, `demographics.csv` has 4 categories of 3000 patients that tell us more about the patient.<br>
- It has the
    - gender of patient,
    - race of patient,
    - resident status of patient and
    - date of birth of patient<br>
    
- The `patient_id` column of the dataset seems to be the `patient_id` column in the `bill_id_df` datset.<br>
- There are no null values.<br>
- The dataset needs to be cleaned as there are
     - dates in object type<br>
     - synonymous terms representing gender, race & citizenship.<br>
     
Considering the `clinical_data.csv` had 3400 patients' clinical data, the `demographics.csv` only has demogrpahic data for 88% of these patients. 

---
I will clean the datasets for our analysis later on.

### (B) Clean the datasets
`bill_amt` dataset does not require any cleaning.<br>
`bill_id_df` dataset column `date_of_admission` needs to be changed to datetime object.<br>
`data` dataset column `medical_history_3` has string objects that needs to be changed to binary & dates to be changed to datetime objects.<br>
`demographic` dataset column for gender, race and citizenship needs to be changed for consistency & dates to be changed to datetime objects.<br>

---
B1 - Clean `bill_id_df`

In [22]:
bill_id_df['date_of_admission'] = pd.to_datetime(bill_id_df.date_of_admission)

In [23]:
bill_id_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13600 entries, 0 to 13599
Data columns (total 3 columns):
bill_id              13600 non-null int64
patient_id           13600 non-null object
date_of_admission    13600 non-null datetime64[ns]
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 318.8+ KB



---
B2 - Clean `data`

In [24]:
data['date_of_admission'] = pd.to_datetime(data.date_of_admission)

In [25]:
data['date_of_discharge'] = pd.to_datetime(data.date_of_discharge)

In [26]:
data.medical_history_3.unique()

array(['0', 'No', '1', 'Yes'], dtype=object)

In [27]:
data.medical_history_3.value_counts()

0      2176
No      761
1       348
Yes     115
Name: medical_history_3, dtype: int64

In [28]:
data[data['medical_history_3'] == 'No'].head()

id date_of_admission date_of_discharge  \
3   e0397dd72caf4552c5babebd3d61736c        2011-01-02        2011-01-14   
4   94ade3cd5f66f4584902554dff170a29        2011-01-08        2011-01-16   
5   59e07adc2dbc5f70131f57d003610d74        2011-01-07        2011-01-17   
8   457402b26562d41f4e40906d3d17d5d1        2011-01-12        2011-01-18   
10  d45fcf3bec63ae3b436be42fad33b9e0        2011-01-12        2011-01-18   

    medical_history_1  medical_history_2 medical_history_3  medical_history_4  \
3                   0                1.0                No                  0   
4                   0                0.0                No                  0   
5                   0                NaN                No                  0   
8                   0                0.0                No                  0   
10                  1                NaN                No                  1   

    medical_history_5  medical_history_6  medical_history_7  \
3                 0.0                  1                  1   
4                 0.0                  1                  1   
5                 NaN                  0                  0   
8                 0.0                  0                  0   
10                0.0                  0                  0   

    preop_medication_1  preop_medication_2  preop_medication_3  \
3                    1                   0                   1   
4                    0                   0                   0   
5                    0                   1                   1   
8                    0                   1                   1   
10                   0                   1                   1   

    preop_medication_4  preop_medication_5  preop_medication_6  symptom_1  \
3                    0                   0                   1          1   
4                    0                   1                   0          0   
5                    1                   1                   1          1   
8                    1                   1                   0          1   
10                   1                   1                   1          1   

    symptom_2  symptom_3  symptom_4  symptom_5  lab_result_1  lab_result_2  \
3           1          1          1          1          13.3          28.4   
4           1          0          1          0          12.0          27.8   
5           0          1          1          1          15.8          31.0   
8           1          1          0          0          12.5          32.9   
10          1          0          1          0          17.0          27.7   

    lab_result_3  weight  height  
3           76.0    64.4   152.0  
4           87.0    55.6   160.0  
5           75.0    78.8   169.0  
8           87.0    98.4   166.0  
10         116.0    91.6   169.0

In [29]:
data['medical_history_3'] = data['medical_history_3'].map(lambda x: 0.0 if x.strip() == 'No' else 1.0 if x == 'Yes' else float(x.strip()))

In [30]:
data.medical_history_3.value_counts()

0.0    2937
1.0     463
Name: medical_history_3, dtype: int64

In [31]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3400 entries, 0 to 3399
Data columns (total 26 columns):
id                    3400 non-null object
date_of_admission     3400 non-null datetime64[ns]
date_of_discharge     3400 non-null datetime64[ns]
medical_history_1     3400 non-null int64
medical_history_2     3167 non-null float64
medical_history_3     3400 non-null float64
medical_history_4     3400 non-null int64
medical_history_5     3096 non-null float64
medical_history_6     3400 non-null int64
medical_history_7     3400 non-null int64
preop_medication_1    3400 non-null int64
preop_medication_2    3400 non-null int64
preop_medication_3    3400 non-null int64
preop_medication_4    3400 non-null int64
preop_medication_5    3400 non-null int64
preop_medication_6    3400 non-null int64
symptom_1             3400 non-null int64
symptom_2             3400 non-null int64
symptom_3             3400 non-null int64
symptom_4             3400 non-null int64
symptom_5             3400 n

In [32]:
data.head()

id date_of_admission date_of_discharge  \
0  1d21f2be18683991eb93d182d6b2d220        2011-01-01        2011-01-11   
1  62bdca0b95d97e99e1c712048fb9fd09        2011-01-01        2011-01-11   
2  c85cf97bc6307ded0dd4fef8bad2fa09        2011-01-02        2011-01-13   
3  e0397dd72caf4552c5babebd3d61736c        2011-01-02        2011-01-14   
4  94ade3cd5f66f4584902554dff170a29        2011-01-08        2011-01-16   

   medical_history_1  medical_history_2  medical_history_3  medical_history_4  \
0                  0                1.0                0.0                  0   
1                  0                0.0                0.0                  0   
2                  0                0.0                0.0                  0   
3                  0                1.0                0.0                  0   
4                  0                0.0                0.0                  0   

   medical_history_5  medical_history_6  medical_history_7  \
0                0.0                  0                  0   
1                0.0                  0                  0   
2                0.0                  0                  0   
3                0.0                  1                  1   
4                0.0                  1                  1   

   preop_medication_1  preop_medication_2  preop_medication_3  \
0                   1                   0                   1   
1                   0                   1                   1   
2                   0                   1                   1   
3                   1                   0                   1   
4                   0                   0                   0   

   preop_medication_4  preop_medication_5  preop_medication_6  symptom_1  \
0                   0                   0                   1          0   
1                   1                   1                   0          0   
2                   1                   1                   1          1   
3                   0                   0                   1          1   
4                   0                   1                   0          0   

   symptom_2  symptom_3  symptom_4  symptom_5  lab_result_1  lab_result_2  \
0          0          0          1          1          13.2          30.9   
1          0          1          1          1          13.8          22.6   
2          1          1          1          0          11.2          26.2   
3          1          1          1          1          13.3          28.4   
4          1          0          1          0          12.0          27.8   

   lab_result_3  weight  height  
0         123.0    71.3   161.0  
1          89.0    78.4   160.0  
2         100.0    72.0   151.0  
3          76.0    64.4   152.0  
4          87.0    55.6   160.0

In [33]:
data.rename(index=str, columns={"id": "patient_id"}, inplace= True)

In [34]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3400 entries, 0 to 3399
Data columns (total 26 columns):
patient_id            3400 non-null object
date_of_admission     3400 non-null datetime64[ns]
date_of_discharge     3400 non-null datetime64[ns]
medical_history_1     3400 non-null int64
medical_history_2     3167 non-null float64
medical_history_3     3400 non-null float64
medical_history_4     3400 non-null int64
medical_history_5     3096 non-null float64
medical_history_6     3400 non-null int64
medical_history_7     3400 non-null int64
preop_medication_1    3400 non-null int64
preop_medication_2    3400 non-null int64
preop_medication_3    3400 non-null int64
preop_medication_4    3400 non-null int64
preop_medication_5    3400 non-null int64
preop_medication_6    3400 non-null int64
symptom_1             3400 non-null int64
symptom_2             3400 non-null int64
symptom_3             3400 non-null int64
symptom_4             3400 non-null int64
symptom_5             3400 non-nu


---
B3 - Clean `demographic`

In [35]:
demographic['date_of_birth'] = pd.to_datetime(demographic.date_of_birth)

In [36]:
demographic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 5 columns):
patient_id         3000 non-null object
gender             3000 non-null object
race               3000 non-null object
resident_status    3000 non-null object
date_of_birth      3000 non-null datetime64[ns]
dtypes: datetime64[ns](1), object(4)
memory usage: 117.3+ KB


In [37]:
demographic.race.unique()

array(['Indian', 'Chinese', 'Malay', 'chinese', 'India', 'Others'],
      dtype=object)

In [38]:
demographic.patient_id.nunique()

3000

In [39]:
demographic.race.value_counts()

Chinese    1608
Malay       629
chinese     307
Indian      195
Others      161
India       100
Name: race, dtype: int64

In [40]:
demographic['race'] = demographic['race'].map(lambda x: 'Chinese' if x.strip() == 'chinese' else 'Indian' if x == 'India' else str(x.strip()))

In [41]:
demographic.race.value_counts()

Chinese    1915
Malay       629
Indian      295
Others      161
Name: race, dtype: int64

In [42]:
demographic.resident_status.unique()

array(['Singaporean', 'Singapore citizen', 'PR', 'Foreigner'],
      dtype=object)

In [43]:
demographic.resident_status.value_counts()

Singaporean          1782
Singapore citizen     610
PR                    465
Foreigner             143
Name: resident_status, dtype: int64

In [44]:
demographic['resident_status'] = demographic['resident_status'].map(lambda x: 'Singaporean' if x.strip() == 'Singapore citizen' else str(x.strip()))

In [45]:
demographic.resident_status.value_counts()

Singaporean    2392
PR              465
Foreigner       143
Name: resident_status, dtype: int64

In [46]:
demographic.gender.unique()

array(['Female', 'f', 'Male', 'm'], dtype=object)

In [47]:
demographic.gender.value_counts()

Female    1396
Male      1333
m          170
f          101
Name: gender, dtype: int64

In [48]:
demographic['gender'] = demographic['gender'].map(lambda x: 'Female' if x.strip() == 'f' else 'Male' if x == 'm' else str(x.strip()))

In [49]:
demographic.gender.value_counts()

Male      1503
Female    1497
Name: gender, dtype: int64

---
B4 - Save the cleaned in data to the cleaned_dataset folder

In [50]:
# bill_id_df.to_csv('clean_datasets/bill_id.csv', sep=',')

In [51]:
# data.to_csv('clean_datasets/clinical_data.csv', sep=',')

In [52]:
# demographic.to_csv('clean_datasets/demographics.csv', sep=',')

### (C) Join the datasets

From the problem at hand, we know that we are analysing insights of the factors that drives cost for patients sharing the same condition. The patients' personal details, medical bill and report data has been provided. As cost is the main focus, we need to combine the dataset so that it shows all the different features that results in the cost.
- Combine `bill_id_df` to `bill_amt`, using the bill_id column & save it as bill data,
- Combine `demographic` to `data`, using the id column to patient_id column & save it as patients' data,
- Combine patients' data to bill, using the id column & save it as patients' cost data.

---
C1 - Join `bill_id_df` & `bill_amt`

In [53]:
bill = pd.merge(bill_id_df, bill_amt, on= 'bill_id')

In [54]:
bill.head()

bill_id                        patient_id date_of_admission  \
0  7968360812  1d21f2be18683991eb93d182d6b2d220        2011-01-01   
1  6180579974  62bdca0b95d97e99e1c712048fb9fd09        2011-01-01   
2  7512568183  1d21f2be18683991eb93d182d6b2d220        2011-01-01   
3  3762633379  62bdca0b95d97e99e1c712048fb9fd09        2011-01-01   
4  7654730355  1d21f2be18683991eb93d182d6b2d220        2011-01-01   

         amount  
0    951.214467  
1  19900.297086  
2  12338.269972  
3   1145.372193  
4   1133.601902

In [55]:
bill[bill['patient_id'] == '4e46fddfa404b306809c350aecbf0f6a']

bill_id                        patient_id date_of_admission  \
2517   839597397  4e46fddfa404b306809c350aecbf0f6a        2011-11-23   
2519  1658432711  4e46fddfa404b306809c350aecbf0f6a        2011-11-23   
2520  1808393173  4e46fddfa404b306809c350aecbf0f6a        2011-11-23   
2523  9525586787  4e46fddfa404b306809c350aecbf0f6a        2011-11-23   
5433   543243535  4e46fddfa404b306809c350aecbf0f6a        2013-01-06   
5435  1288390292  4e46fddfa404b306809c350aecbf0f6a        2013-01-06   
5436  9600908611  4e46fddfa404b306809c350aecbf0f6a        2013-01-06   
5437  7875201153  4e46fddfa404b306809c350aecbf0f6a        2013-01-06   
5564   118453328  4e46fddfa404b306809c350aecbf0f6a        2013-01-23   
5567  3680165171  4e46fddfa404b306809c350aecbf0f6a        2013-01-23   
5568   136013293  4e46fddfa404b306809c350aecbf0f6a        2013-01-23   
5570  4538054359  4e46fddfa404b306809c350aecbf0f6a        2013-01-23   
7298   483325687  4e46fddfa404b306809c350aecbf0f6a        2013-09-16   
7301  3619424812  4e46fddfa404b306809c350aecbf0f6a        2013-09-16   
7304  5045123685  4e46fddfa404b306809c350aecbf0f6a        2013-09-16   
7307  5694828317  4e46fddfa404b306809c350aecbf0f6a        2013-09-16   

            amount  
2517   6132.750887  
2519    868.435533  
2520  10361.862067  
2523   1354.232142  
5433    769.098299  
5435   7675.021417  
5436   1269.881609  
5437    797.989644  
5564   1557.500102  
5567    592.684082  
5568    746.477367  
5570   2053.144782  
7298   1487.709021  
7301    975.982803  
7304  16321.969490  
7307   9242.658865

In [56]:
bill = bill.groupby(['patient_id', 'date_of_admission']).sum().sum(
    level=['patient_id', 'date_of_admission']).drop(['bill_id'], axis = 1).reset_index()


In [57]:
bill[bill['patient_id'] == '4e46fddfa404b306809c350aecbf0f6a']

patient_id date_of_admission        amount
1024  4e46fddfa404b306809c350aecbf0f6a        2011-11-23  18717.280628
1025  4e46fddfa404b306809c350aecbf0f6a        2013-01-06  10511.990970
1026  4e46fddfa404b306809c350aecbf0f6a        2013-01-23   4949.806331
1027  4e46fddfa404b306809c350aecbf0f6a        2013-09-16  28028.320179

In [58]:
bill.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3400 entries, 0 to 3399
Data columns (total 3 columns):
patient_id           3400 non-null object
date_of_admission    3400 non-null datetime64[ns]
amount               3400 non-null float64
dtypes: datetime64[ns](1), float64(1), object(1)
memory usage: 79.8+ KB


In [59]:
# bill.to_csv('clean_datasets/bill.csv', sep=',')

---
C2 - Join `demographic` & `data`

In [60]:
patient_data = pd.merge(demographic, data, on= 'patient_id')

In [61]:
patient_data.head()

patient_id  gender     race resident_status  \
0  fa2d818b2261e44e30628ad1ac9cc72c  Female   Indian     Singaporean   
1  5b6477c5de78d0b138e3b0c18e21d0ae  Female  Chinese     Singaporean   
2  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
3  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
4  c7f3881684045e6c49020481020fae36    Male    Malay     Singaporean   

  date_of_birth date_of_admission date_of_discharge  medical_history_1  \
0    1971-05-14        2014-09-20        2014-10-04                  0   
1    1976-02-18        2014-08-08        2014-08-18                  0   
2    1982-07-03        2011-03-10        2011-03-20                  0   
3    1982-07-03        2013-07-02        2013-07-14                  0   
4    1947-06-15        2015-10-28        2015-11-10                  0   

   medical_history_2  medical_history_3  medical_history_4  medical_history_5  \
0                0.0                0.0                  0                0.0   
1                0.0                0.0                  0                NaN   
2                0.0                0.0                  0                0.0   
3                0.0                0.0                  1                NaN   
4                1.0                0.0                  0                0.0   

   medical_history_6  medical_history_7  preop_medication_1  \
0                  0                  1                   1   
1                  0                  1                   0   
2                  1                  0                   0   
3                  0                  0                   1   
4                  0                  0                   0   

   preop_medication_2  preop_medication_3  preop_medication_4  \
0                   0                   1                   0   
1                   0                   0                   1   
2                   1                   1                   1   
3                   1                   0                   1   
4                   1                   0                   0   

   preop_medication_5  preop_medication_6  symptom_1  symptom_2  symptom_3  \
0                   1                   1          1          1          0   
1                   1                   1          1          1          1   
2                   1                   0          0          0          1   
3                   0                   1          1          0          0   
4                   1                   1          1          1          1   

   symptom_4  symptom_5  lab_result_1  lab_result_2  lab_result_3  weight  \
0          0          1          14.2          27.0          81.0    68.1   
1          1          1          14.3          27.2          92.0    76.5   
2          1          1          14.4          26.6          92.0    69.7   
3          1          0          18.0          26.6          92.0    66.7   
4          1          1          14.9          26.7          97.0    96.9   

   height  
0   161.0  
1   151.0  
2   175.0  
3   175.0  
4   180.0

In [62]:
patient_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3400 entries, 0 to 3399
Data columns (total 30 columns):
patient_id            3400 non-null object
gender                3400 non-null object
race                  3400 non-null object
resident_status       3400 non-null object
date_of_birth         3400 non-null datetime64[ns]
date_of_admission     3400 non-null datetime64[ns]
date_of_discharge     3400 non-null datetime64[ns]
medical_history_1     3400 non-null int64
medical_history_2     3167 non-null float64
medical_history_3     3400 non-null float64
medical_history_4     3400 non-null int64
medical_history_5     3096 non-null float64
medical_history_6     3400 non-null int64
medical_history_7     3400 non-null int64
preop_medication_1    3400 non-null int64
preop_medication_2    3400 non-null int64
preop_medication_3    3400 non-null int64
preop_medication_4    3400 non-null int64
preop_medication_5    3400 non-null int64
preop_medication_6    3400 non-null int64
symptom_1       

In [63]:
patient_data.patient_id.value_counts()

4e46fddfa404b306809c350aecbf0f6a    4
b2d15cda8c4e1f86ba43356434df6718    3
f01974d944d1c4b2c40753223fb979fa    3
e3270f450ee164e596ca933a25bab61d    3
d15ec3b278e42b36e7362ac783643685    3
af713a949eda293f348bf51a058e48a1    3
ea4f259bc46b6818f55b774abb7ac690    3
d01386ff66ee5ecef47c5ef7980ff10a    3
f7914ba012897ceb6562be47f6cec84d    3
a0214fcb0b1e3cf8d667e77a27f9840c    3
86cbe18dab50db6269046fc539915636    3
cebd42e84733dae9898687cfb750fbaf    3
cf835526e33d00afa016ac898582038c    3
6a6991ecb85a2f82f77b09eb0eeb747b    3
258807316af4b45fda1b05668d557d06    3
8239986dfdf8f4e0bb351ace4742ef95    3
5e9e8508e8098fc220a12db23c698ec6    3
64a531972193ccf232cc47597ddb85ed    3
0eacfb2daed1f3ba2adf32e293bc05a6    3
feb52392f4d58a02254cbd0ba634312f    3
d3a8ec14276d1b7d1f25214cbc979dcb    2
221210f6c814b574ab81931cbbb16742    2
8f2f9f57db98456977d660831f674472    2
276be5fbec1f8503c2f159b1cf827ea0    2
a775d45d2bd24b636db1541003c6a260    2
ac634c817614a1256ecf272d8413cda8    2
15afd10b6cbb

`patient_data` has data of 3000 unique patients, with patients who have been admitted for 1 to 4 times. Only 1 patient has been admitted for 4 times and most have been admitted only once.<br>
<br>
Let's add the frequency of being admitted in the dataset.

In [64]:
patient_data['freq']= patient_data.groupby('patient_id')['patient_id'].transform('count')

In [65]:
patient_data.head()

patient_id  gender     race resident_status  \
0  fa2d818b2261e44e30628ad1ac9cc72c  Female   Indian     Singaporean   
1  5b6477c5de78d0b138e3b0c18e21d0ae  Female  Chinese     Singaporean   
2  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
3  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
4  c7f3881684045e6c49020481020fae36    Male    Malay     Singaporean   

  date_of_birth date_of_admission date_of_discharge  medical_history_1  \
0    1971-05-14        2014-09-20        2014-10-04                  0   
1    1976-02-18        2014-08-08        2014-08-18                  0   
2    1982-07-03        2011-03-10        2011-03-20                  0   
3    1982-07-03        2013-07-02        2013-07-14                  0   
4    1947-06-15        2015-10-28        2015-11-10                  0   

   medical_history_2  medical_history_3  medical_history_4  medical_history_5  \
0                0.0                0.0                  0                0.0   
1                0.0                0.0                  0                NaN   
2                0.0                0.0                  0                0.0   
3                0.0                0.0                  1                NaN   
4                1.0                0.0                  0                0.0   

   medical_history_6  medical_history_7  preop_medication_1  \
0                  0                  1                   1   
1                  0                  1                   0   
2                  1                  0                   0   
3                  0                  0                   1   
4                  0                  0                   0   

   preop_medication_2  preop_medication_3  preop_medication_4  \
0                   0                   1                   0   
1                   0                   0                   1   
2                   1                   1                   1   
3                   1                   0                   1   
4                   1                   0                   0   

   preop_medication_5  preop_medication_6  symptom_1  symptom_2  symptom_3  \
0                   1                   1          1          1          0   
1                   1                   1          1          1          1   
2                   1                   0          0          0          1   
3                   0                   1          1          0          0   
4                   1                   1          1          1          1   

   symptom_4  symptom_5  lab_result_1  lab_result_2  lab_result_3  weight  \
0          0          1          14.2          27.0          81.0    68.1   
1          1          1          14.3          27.2          92.0    76.5   
2          1          1          14.4          26.6          92.0    69.7   
3          1          0          18.0          26.6          92.0    66.7   
4          1          1          14.9          26.7          97.0    96.9   

   height  freq  
0   161.0     1  
1   151.0     1  
2   175.0     2  
3   175.0     2  
4   180.0     1

In [66]:
patient_data[patient_data['patient_id'] == '4e46fddfa404b306809c350aecbf0f6a']

patient_id gender     race resident_status  \
1113  4e46fddfa404b306809c350aecbf0f6a   Male  Chinese     Singaporean   
1114  4e46fddfa404b306809c350aecbf0f6a   Male  Chinese     Singaporean   
1115  4e46fddfa404b306809c350aecbf0f6a   Male  Chinese     Singaporean   
1116  4e46fddfa404b306809c350aecbf0f6a   Male  Chinese     Singaporean   

     date_of_birth date_of_admission date_of_discharge  medical_history_1  \
1113    1948-06-16        2011-11-23        2011-12-05                  1   
1114    1948-06-16        2013-01-06        2013-01-17                  0   
1115    1948-06-16        2013-01-23        2013-02-02                  0   
1116    1948-06-16        2013-09-16        2013-09-27                  1   

      medical_history_2  medical_history_3  medical_history_4  \
1113                1.0                0.0                  0   
1114                0.0                0.0                  0   
1115                0.0                0.0                  0   
1116                1.0                0.0                  0   

      medical_history_5  medical_history_6  medical_history_7  \
1113                0.0                  0                  0   
1114                0.0                  0                  0   
1115                0.0                  0                  1   
1116                0.0                  1                  0   

      preop_medication_1  preop_medication_2  preop_medication_3  \
1113                   1                   1                   1   
1114                   0                   1                   1   
1115                   0                   0                   1   
1116                   0                   1                   1   

      preop_medication_4  preop_medication_5  preop_medication_6  symptom_1  \
1113                   0                   0                   1          1   
1114                   1                   1                   1          1   
1115                   1                   0                   1          0   
1116                   0                   1                   0          1   

      symptom_2  symptom_3  symptom_4  symptom_5  lab_result_1  lab_result_2  \
1113          0          1          1          0          16.1          25.5   
1114          0          1          0          0          14.1          24.1   
1115          0          0          0          0          13.0          27.0   
1116          1          0          0          1          11.7          22.6   

      lab_result_3  weight  height  freq  
1113         111.0    71.4   166.0     4  
1114         103.0    71.4   166.0     4  
1115          91.0    70.4   166.0     4  
1116          84.0    70.4   166.0     4

In [67]:
patient_data['freq'].value_counts()

1    2621
2     718
3      57
4       4
Name: freq, dtype: int64

Most have been hospitalized for 1 day only.

---

In [68]:
patient_data['days_admitted'] = patient_data.date_of_discharge - patient_data.date_of_admission

In [69]:
patient_data.head()

patient_id  gender     race resident_status  \
0  fa2d818b2261e44e30628ad1ac9cc72c  Female   Indian     Singaporean   
1  5b6477c5de78d0b138e3b0c18e21d0ae  Female  Chinese     Singaporean   
2  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
3  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
4  c7f3881684045e6c49020481020fae36    Male    Malay     Singaporean   

  date_of_birth date_of_admission date_of_discharge  medical_history_1  \
0    1971-05-14        2014-09-20        2014-10-04                  0   
1    1976-02-18        2014-08-08        2014-08-18                  0   
2    1982-07-03        2011-03-10        2011-03-20                  0   
3    1982-07-03        2013-07-02        2013-07-14                  0   
4    1947-06-15        2015-10-28        2015-11-10                  0   

   medical_history_2  medical_history_3  medical_history_4  medical_history_5  \
0                0.0                0.0                  0                0.0   
1                0.0                0.0                  0                NaN   
2                0.0                0.0                  0                0.0   
3                0.0                0.0                  1                NaN   
4                1.0                0.0                  0                0.0   

   medical_history_6  medical_history_7  preop_medication_1  \
0                  0                  1                   1   
1                  0                  1                   0   
2                  1                  0                   0   
3                  0                  0                   1   
4                  0                  0                   0   

   preop_medication_2  preop_medication_3  preop_medication_4  \
0                   0                   1                   0   
1                   0                   0                   1   
2                   1                   1                   1   
3                   1                   0                   1   
4                   1                   0                   0   

   preop_medication_5  preop_medication_6  symptom_1  symptom_2  symptom_3  \
0                   1                   1          1          1          0   
1                   1                   1          1          1          1   
2                   1                   0          0          0          1   
3                   0                   1          1          0          0   
4                   1                   1          1          1          1   

   symptom_4  symptom_5  lab_result_1  lab_result_2  lab_result_3  weight  \
0          0          1          14.2          27.0          81.0    68.1   
1          1          1          14.3          27.2          92.0    76.5   
2          1          1          14.4          26.6          92.0    69.7   
3          1          0          18.0          26.6          92.0    66.7   
4          1          1          14.9          26.7          97.0    96.9   

   height  freq days_admitted  
0   161.0     1       14 days  
1   151.0     1       10 days  
2   175.0     2       10 days  
3   175.0     2       12 days  
4   180.0     1       13 days

In [70]:
patient_data['days_admitted'].value_counts()

11 days    437
12 days    414
10 days    412
9 days     402
13 days    389
8 days     309
14 days    269
15 days    227
7 days     166
16 days    113
6 days     111
17 days     52
5 days      44
18 days     21
4 days      15
3 days       8
19 days      6
2 days       2
20 days      2
1 days       1
Name: days_admitted, dtype: int64

Most get hospitalized for 9 to 12 days.

---
Let's assume all our patients are living and none of them have passed on. We will add a column to represent their age.

In [71]:
now = datetime.now()
patient_data['age'] = (now - patient_data.date_of_birth).astype('<m8[Y]')

In [72]:
patient_data.head()

patient_id  gender     race resident_status  \
0  fa2d818b2261e44e30628ad1ac9cc72c  Female   Indian     Singaporean   
1  5b6477c5de78d0b138e3b0c18e21d0ae  Female  Chinese     Singaporean   
2  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
3  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
4  c7f3881684045e6c49020481020fae36    Male    Malay     Singaporean   

  date_of_birth date_of_admission date_of_discharge  medical_history_1  \
0    1971-05-14        2014-09-20        2014-10-04                  0   
1    1976-02-18        2014-08-08        2014-08-18                  0   
2    1982-07-03        2011-03-10        2011-03-20                  0   
3    1982-07-03        2013-07-02        2013-07-14                  0   
4    1947-06-15        2015-10-28        2015-11-10                  0   

   medical_history_2  medical_history_3  medical_history_4  medical_history_5  \
0                0.0                0.0                  0                0.0   
1                0.0                0.0                  0                NaN   
2                0.0                0.0                  0                0.0   
3                0.0                0.0                  1                NaN   
4                1.0                0.0                  0                0.0   

   medical_history_6  medical_history_7  preop_medication_1  \
0                  0                  1                   1   
1                  0                  1                   0   
2                  1                  0                   0   
3                  0                  0                   1   
4                  0                  0                   0   

   preop_medication_2  preop_medication_3  preop_medication_4  \
0                   0                   1                   0   
1                   0                   0                   1   
2                   1                   1                   1   
3                   1                   0                   1   
4                   1                   0                   0   

   preop_medication_5  preop_medication_6  symptom_1  symptom_2  symptom_3  \
0                   1                   1          1          1          0   
1                   1                   1          1          1          1   
2                   1                   0          0          0          1   
3                   0                   1          1          0          0   
4                   1                   1          1          1          1   

   symptom_4  symptom_5  lab_result_1  lab_result_2  lab_result_3  weight  \
0          0          1          14.2          27.0          81.0    68.1   
1          1          1          14.3          27.2          92.0    76.5   
2          1          1          14.4          26.6          92.0    69.7   
3          1          0          18.0          26.6          92.0    66.7   
4          1          1          14.9          26.7          97.0    96.9   

   height  freq days_admitted   age  
0   161.0     1       14 days  47.0  
1   151.0     1       10 days  42.0  
2   175.0     2       10 days  36.0  
3   175.0     2       12 days  36.0  
4   180.0     1       13 days  71.0

In [73]:
patient_data['age'].value_counts()

43.0    124
45.0    111
44.0    104
46.0    101
48.0     93
41.0     91
65.0     89
39.0     84
47.0     83
36.0     83
42.0     80
40.0     79
70.0     79
49.0     75
50.0     75
57.0     70
66.0     70
51.0     65
52.0     65
38.0     65
68.0     64
74.0     64
53.0     64
76.0     64
72.0     64
73.0     63
58.0     63
77.0     62
54.0     61
69.0     59
55.0     59
63.0     59
79.0     57
56.0     57
75.0     55
62.0     54
60.0     54
37.0     53
59.0     51
71.0     49
67.0     48
83.0     47
82.0     44
78.0     43
64.0     43
61.0     43
80.0     43
84.0     41
81.0     40
35.0     38
34.0     26
85.0     24
86.0     17
87.0     13
33.0     10
32.0      8
88.0      3
28.0      2
31.0      2
27.0      1
29.0      1
30.0      1
Name: age, dtype: int64

In [74]:
# patient_data.to_csv('clean_datasets/patients_data.csv', sep=',')

Most of the patients are 43 to 46 years old.

---
C3 - Join patients data and the corresponding bill

In [75]:
patients = pd.merge(patient_data, bill, on= ['patient_id', 'date_of_admission'])

In [76]:
patients[patients['patient_id'] == '4e46fddfa404b306809c350aecbf0f6a']

patient_id gender     race resident_status  \
1113  4e46fddfa404b306809c350aecbf0f6a   Male  Chinese     Singaporean   
1114  4e46fddfa404b306809c350aecbf0f6a   Male  Chinese     Singaporean   
1115  4e46fddfa404b306809c350aecbf0f6a   Male  Chinese     Singaporean   
1116  4e46fddfa404b306809c350aecbf0f6a   Male  Chinese     Singaporean   

     date_of_birth date_of_admission date_of_discharge  medical_history_1  \
1113    1948-06-16        2011-11-23        2011-12-05                  1   
1114    1948-06-16        2013-01-06        2013-01-17                  0   
1115    1948-06-16        2013-01-23        2013-02-02                  0   
1116    1948-06-16        2013-09-16        2013-09-27                  1   

      medical_history_2  medical_history_3  medical_history_4  \
1113                1.0                0.0                  0   
1114                0.0                0.0                  0   
1115                0.0                0.0                  0   
1116                1.0                0.0                  0   

      medical_history_5  medical_history_6  medical_history_7  \
1113                0.0                  0                  0   
1114                0.0                  0                  0   
1115                0.0                  0                  1   
1116                0.0                  1                  0   

      preop_medication_1  preop_medication_2  preop_medication_3  \
1113                   1                   1                   1   
1114                   0                   1                   1   
1115                   0                   0                   1   
1116                   0                   1                   1   

      preop_medication_4  preop_medication_5  preop_medication_6  symptom_1  \
1113                   0                   0                   1          1   
1114                   1                   1                   1          1   
1115                   1                   0                   1          0   
1116                   0                   1                   0          1   

      symptom_2  symptom_3  symptom_4  symptom_5  lab_result_1  lab_result_2  \
1113          0          1          1          0          16.1          25.5   
1114          0          1          0          0          14.1          24.1   
1115          0          0          0          0          13.0          27.0   
1116          1          0          0          1          11.7          22.6   

      lab_result_3  weight  height  freq days_admitted   age        amount  
1113         111.0    71.4   166.0     4       12 days  70.0  18717.280628  
1114         103.0    71.4   166.0     4       11 days  70.0  10511.990970  
1115          91.0    70.4   166.0     4       10 days  70.0   4949.806331  
1116          84.0    70.4   166.0     4       11 days  70.0  28028.320179

The bill amount can be round to 2 decimal places so that it reflects the price paid by the patients

In [77]:
patients = patients.round({'amount': 2})

In [78]:
patients.head()

patient_id  gender     race resident_status  \
0  fa2d818b2261e44e30628ad1ac9cc72c  Female   Indian     Singaporean   
1  5b6477c5de78d0b138e3b0c18e21d0ae  Female  Chinese     Singaporean   
2  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
3  320aa16c61937447fd6631bf635e7fde    Male  Chinese     Singaporean   
4  c7f3881684045e6c49020481020fae36    Male    Malay     Singaporean   

  date_of_birth date_of_admission date_of_discharge  medical_history_1  \
0    1971-05-14        2014-09-20        2014-10-04                  0   
1    1976-02-18        2014-08-08        2014-08-18                  0   
2    1982-07-03        2011-03-10        2011-03-20                  0   
3    1982-07-03        2013-07-02        2013-07-14                  0   
4    1947-06-15        2015-10-28        2015-11-10                  0   

   medical_history_2  medical_history_3  medical_history_4  medical_history_5  \
0                0.0                0.0                  0                0.0   
1                0.0                0.0                  0                NaN   
2                0.0                0.0                  0                0.0   
3                0.0                0.0                  1                NaN   
4                1.0                0.0                  0                0.0   

   medical_history_6  medical_history_7  preop_medication_1  \
0                  0                  1                   1   
1                  0                  1                   0   
2                  1                  0                   0   
3                  0                  0                   1   
4                  0                  0                   0   

   preop_medication_2  preop_medication_3  preop_medication_4  \
0                   0                   1                   0   
1                   0                   0                   1   
2                   1                   1                   1   
3                   1                   0                   1   
4                   1                   0                   0   

   preop_medication_5  preop_medication_6  symptom_1  symptom_2  symptom_3  \
0                   1                   1          1          1          0   
1                   1                   1          1          1          1   
2                   1                   0          0          0          1   
3                   0                   1          1          0          0   
4                   1                   1          1          1          1   

   symptom_4  symptom_5  lab_result_1  lab_result_2  lab_result_3  weight  \
0          0          1          14.2          27.0          81.0    68.1   
1          1          1          14.3          27.2          92.0    76.5   
2          1          1          14.4          26.6          92.0    69.7   
3          1          0          18.0          26.6          92.0    66.7   
4          1          1          14.9          26.7          97.0    96.9   

   height  freq days_admitted   age    amount  
0   161.0     1       14 days  47.0  19564.17  
1   151.0     1       10 days  42.0  20465.40  
2   175.0     2       10 days  36.0  14435.91  
3   175.0     2       12 days  36.0   6578.39  
4   180.0     1       13 days  71.0  45452.31

In [79]:
# patients.to_csv('clean_datasets/patients.csv', sep=',')

The `patients` dataset has all the combined information of the costs incurred by a patient for each of the time they were admitted. Now we are ready to analyse our data.